Logging configuration fix

azax42 opened the pull request:
Hello, my company has adopted your package for our internal use and I have made some enhancements that I think might be of use outside of our company.

This PR allows adjusting the log level via a cli switch or environment variable, cleans up the log message format and fixes an ambiguous log message 

If you are willing to accept this PR (or PRs in general) I plan to submit a few more PRs of roughly similar size and scope

azax42 committed:
Add `--log-level` cli switch and `LOG_LEVEL` enviornment variable to control log level

azax42 committed:
Simplify code and cleanup message format using `logging.basicConfig()`

azax42 committed:
Make "Total migrations to apply:" refer to unapplied migrations

zifter commented on src/clickhouse_migrations/cmd.py:
```diff
@@ -13,6 +14,11 @@
 )
 
+def log_level(value: str) -> str:
+    if value.upper() in ("DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"):
```
maybe it's better to use logging._nameToLevel.keys()?
yes, it's private variable, but it contains all levels and aliases.

zifter commented:
You need to add small unit test, which uses log_level :)
I'm trying to keep 100% coverage.

zifter commented:
@azax42 Thank you very much for your PR!
I'm ready to accept, review and discuss any changes to this repo, so feel free to make more PR! :)

azax42 committed:
Cleanup unused import

azax42 committed:
Use `logging._nameToLevel` for list of valid log levels

azax42 committed:
Formatting changes

azax42 committed:
Use public api when available

azax42 committed:
Reformating and minor refactroing to pass tests

azax42 committed:
Disable W0212 pylint check

private api used only if public api is not available (old python versions) so we don't have to care if  `logging._nameToLevel` goes away in the future

azax42 committed:
Add tests for `log_level()`

azax42 committed:
Make combined multi version coverage tests work

zifter commented on src/tests/test_log_level.py:
```diff
@@ -0,0 +1,48 @@
+
+
+def test_invalid_log_levels():
+    bad_input = (
```
it's good to use parameterized for table tests :)

azax42 replied:
Not sure I understand?
